src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageMetadataReadMeControl.xaml.cs

```diff
@@ -0,0 +1,160 @@
+
+namespace NuGet.PackageManagement.UI
+{
+
+    /// <summary>
+    /// Interaction logic for PackageMetadataReadMeControl.xaml
+    /// </summary>
+    public partial class PackageMetadataReadMeControl : UserControl, IDisposable
+    {
+        public static readonly DependencyProperty PackageMetadataProperty =
+            DependencyProperty.Register(
+                nameof(PackageMetadata),
+                typeof(DetailedPackageMetadata),
+                typeof(PackageMetadataReadMeControl),
+                new PropertyMetadata(OnPropertyChanged));
+
+#pragma warning disable CS0618 // Type or member is obsolete
+        private IMarkdownPreview _markdownPreview;
+#pragma warning restore CS0618 // Type or member is obsolete
+        private bool _disposed = false;
+
+        private ReadMePreviewViewModel ReadMeViewModel { get => (ReadMePreviewViewModel)DataContext; }
+
+        [Obsolete]
+        public PackageMetadataReadMeControl()
+        {
+            InitializeComponent();
+        }
+
+        public void Dispose()
+        {
+            Dispose(disposing: true);
+            GC.SuppressFinalize(this);
+        }
+
+        public DetailedPackageMetadata PackageMetadata
+        {
+            get
+            {
+                return (DetailedPackageMetadata)GetValue(PackageMetadataProperty);
+            }
+            set
+            {
+                UpdateControl(PackageMetadata, value);
+                SetValue(PackageMetadataProperty, value);
+            }
+        }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (_disposed)
+            {
+                return;
+            }
+
+            if (disposing)
+            {
+                _markdownPreview?.Dispose();
+                var viewModel = (ReadMePreviewViewModel)DataContext;
+                viewModel.PropertyChanged -= ViewModel_PropertyChangedAsync;
+            }
+
+            _disposed = true;
+        }
+
+        private static void OnPropertyChanged(
+           DependencyObject dependencyObject,
+           DependencyPropertyChangedEventArgs e)
+        {
+            var control = (PackageMetadataReadMeControl)dependencyObject;
+            if (e.Property == PackageMetadataProperty)
+            {
+                control?.UpdateControl((DetailedPackageMetadata)e.OldValue, (DetailedPackageMetadata)e.NewValue);
+            }
+        }
+
+        private void UpdateControl(DetailedPackageMetadata oldValue, DetailedPackageMetadata newValue)
+        {
+            if (newValue is not null &&
+                (oldValue?.Id != newValue.Id ||
+                oldValue?.PackagePath != newValue.PackagePath))
+            {
+                NuGetUIThreadHelper.JoinableTaskFactory
+                .RunAsync(async () =>
+                {
+                    await ReadMeViewModel.LoadReadme(newValue.PackagePath, newValue.Id);
+                })
+                .PostOnFailure(nameof(DetailControlModel));
```

Comment:
Should this just be removed since LoadReadme already catches all and logs fault telemetry?

Reply:
LoadReadme isn't catching errors anymore